src/messages/direct.rs

```diff
@@ -29,15 +32,16 @@ pub enum DirectMessage {
     /// Sent from members of a section or group message's source authority to the first hop. The
     /// message will only be relayed once enough signatures have been accumulated.
     MessageSignature(SignedRoutingMessage),
-    /// Sent from a newly connected peer to the bootstrap node to prove that it is the owner of
-    /// the client's claimed public ID.
-    BootstrapRequest,
+    /// Sent from a newly connected peer to the bootstrap node to request connection infos of
+    /// members of the section matching the given name.
+    BootstrapRequest(XorName),
```

Comment:
Why now need the name here?
Isn't the target section shall expect such request? and if not, then shall reject it?

Reply:
It's to support relocation. A relocating node still has its old id but already needs to bootstrap to the target section which it knows only by name (which was given to it by the source section). The response to this contain the section prefix which the node then uses to generate its new id.